test/SDKMethodRunner_test.py

```diff
@@ -143,96 +141,94 @@ def test_get_module_git_commit(self):
+            self.assertCountEqual(job_input.source_ws_objects, ["a/b/c", "e/d"])
+            self.assertEqual(job_input.parent_job_id, "9998")
 
-        self.assertFalse(result.get("job_output"))
+            self.assertFalse(job.job_output)
 
-        self.test_collection.delete_one({"_id": ObjectId(job_id)})
-        self.assertEqual(self.test_collection.count(), 0)
+            Job.objects.get(id=job_id).delete()
+            self.assertEqual(ori_job_count, Job.objects.count())
 
     def test_get_job_params(self):
 
-        self.assertEqual(self.test_collection.count(), 0)
-        job_id = self.create_job_rec()
-        self.assertEqual(self.test_collection.count(), 1)
+        with self.mongo_util.me_collection(self.cfg["mongo-jobs-collection"]):
+            ori_job_count = Job.objects.count()
+            job_id = self.create_job_rec()
+            self.assertEqual(ori_job_count, Job.objects.count()-1)
 
-        runner = self.getRunner()
-        params = runner.get_job_params(job_id)
+            runner = self.getRunner()
 
-        expected_params_keys = ["wsid", "method", "params", "service_ver", "app_id",
-                                "source_ws_objects", "parent_job_id"]
-        self.assertCountEqual(params.keys(), expected_params_keys)
-        self.assertEqual(params["wsid"], self.ws_id)
-        self.assertEqual(params["method"], "MEGAHIT.run_megahit")
-        self.assertEqual(params["app_id"], "MEGAHIT/run_megahit")
-        self.assertEqual(params["service_ver"], "2.2.1")
-        self.assertCountEqual(params["source_ws_objects"], ["a/b/c", "e/d"])
-        self.assertEqual(params["parent_job_id"], "9998")
+            params = runner.get_job_params(job_id)
 
-        self.test_collection.delete_one({"_id": ObjectId(job_id)})
-        self.assertEqual(self.test_collection.count(), 0)
+            expected_params_keys = ["wsid", "method", "params", "service_ver", "app_id",
+                                    "source_ws_objects", "parent_job_id"]
+            self.assertCountEqual(params.keys(), expected_params_keys)
+            self.assertEqual(params["wsid"], self.ws_id)
+            self.assertEqual(params["method"], "MEGAHIT.run_megahit")
+            self.assertEqual(params["app_id"], "MEGAHIT/run_megahit")
+            self.assertEqual(params["service_ver"], "2.2.1")
+            self.assertCountEqual(params["source_ws_objects"], ["a/b/c", "e/d"])
+            self.assertEqual(params["parent_job_id"], "9998")
+
+            Job.objects.get(id=job_id).delete()
+            self.assertEqual(ori_job_count, Job.objects.count())
 
     def test_update_job_status(self):
 
-        self.assertEqual(self.test_collection.count(), 0)
-        job_id = self.create_job_rec()
-        self.assertEqual(self.test_collection.count(), 1)
+        with self.mongo_util.me_collection(self.cfg["mongo-jobs-collection"]):
 
-        runner = self.getRunner()
+            ori_job_count = Job.objects.count()
```

Comment:
can you please use countDocuments, count is deprecated

Reply:
hmm I found it in latest mongoengine doc. Besides Job.objects.count_documents() is not defined. 

http://docs.mongoengine.org/guide/querying.html#counting-results

Reply:
For pymongo it is deprecated.

https://docs.mongodb.com/manual/reference/method/db.collection.count/

Reply:
You are right, I was not paying enough attention my comment was wrong.